code/modules/mob/living/silicon/robot/emote.dm

```diff
@@ -155,6 +155,23 @@
 			playsound(get_turf(src), 'sound/machines/twobeep.ogg', 50, 0)
 			m_type = VISIBLE
 
+		if("scream")
+			var/M = null
+			if(param)
+				for (var/mob/A in view(null, null))
```

Comment:
It checks if your *[emote]-[target] is in view using some kind of oldcode sorcery in the beginning of the file that just works

![something](https://cloud.githubusercontent.com/assets/17928298/21788096/9cb107cc-d6ab-11e6-8ab7-466ae096272f.png)